From 0ea5cc752bfeeb02748eaafe3864ba44aba317d4 Mon Sep 17 00:00:00 2001
From: Jean-Pierre Geslin <jarsop@outlook.com>
Date: Fri, 27 Mar 2026 21:34:55 +0100
Subject: [PATCH 1/2] use crates.io registry

Upstream-Status: Pending
Signed-off-by: Jean-Pierre Geslin <jarsop@outlook.com>
---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 16adefc..6143b47 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,10 +44,10 @@ maintenance = { status = "actively-developed" }
 [dependencies]
 paste = "1.0.14"
 pyo3 = { version = "0.25.1", features = ["abi3-py39", "extension-module"] }
-zenoh = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", features = [
+zenoh = { version = "1.8.0", features = [
   "internal",
   "unstable",
 ], default-features = false }
-zenoh-ext = { version = "1.8.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.8.0", features = [
+zenoh-ext = { version = "1.8.0", features = [
   "internal",
 ], optional = true }
